src/mixins/yield-erc/ERC20Vault.sol

```diff
@@ -80,13 +80,28 @@ contract ERC4626Vault is ERC20 {
     /// @param to The address to receive underlying tokens corresponding to the withdrawal.
     /// @param underlyingAmount The amount of underlying tokens to withdraw.
     function withdraw(address to, uint256 underlyingAmount) external virtual returns (uint256 shares) {
-        shares = underlyingAmount.fdiv(exchangeRate(), baseUnit);
+        return _withdraw(msg.sender, to, underlyingAmount);
+    }
+
+    /// @notice Withdraw a specific amount of shares for underlying tokens on behalf of `from`.
+    /// @param from The address to redeem shares from.
+    /// @param to The address to receive underlying tokens corresponding to the withdrawal.
+    /// @param shareAmount The amount of shares to redeem for underlying tokens.
+    function withdrawFrom(address from, address to, uint256 shareAmount) external virtual returns (uint256 underlyingAmount) {
+        if (allowance[from][msg.sender] != type(uint256).max) {
+            allowance[from][msg.sender] -= shareAmount;
+        }
+        return _withdraw(from, to, shareAmount);
```

Comment:
`_withdraw` accepts an amount of underlying, not an amount of shares.

I think the third param of `withdrawFrom` should be an `underlyingAmount` rather than a `shareAmount` (so that it is consistent with the `withdraw` and `_withdraw` functions).

Reply:
I suspect this was a copy/paste issue -- copying the `redeemFrom` function but not fully adjusting it correctly.

Reply:
the issue was deeper than that, it had a logic bug where I needed to subtract the share amount from allowance but that hadn't been calculated yet. I reorganized everything to clean this up